data/test_trackers/tracker_moodbot.json

```diff
@@ -37,7 +37,8 @@
     "name": null,
     "requested_slot": null,
     "return_value": null,
-    "session_started_metadata": null
+    "session_started_metadata": null,
+    "counter_utter_ask": 0.0
```

Comment:
I'd prefer this not to be a slot but rather be part of the context. 

Ideally, we port this to `dm2-cleanups` branch and use the frame classes there to add a new attribute to the `CollectInformationPatternFlowStackFrame`.

@ancalita you can see how Thomas modifes the context of a stackframe here: https://github.com/RasaHQ/rasa/blob/eb528753e85084d5a3df53e97631348162924541/rasa/cdu/patterns/clarify.py#L99-L100

Reply:
the variable would then become available as `context.counter_utter_ask`.

and maybe should be renamed to smth like `number_of_reprompts` or something

Reply:
Completely agree with the rename.
Regarding the change to context -> with Thomas' example in mind, is it required to add a new default action `action_collect_information` (in lack of a better name) that could replace the `action_extract_slots` in the `pattern_ask_collect_information`? I was thinking that `action_extract_slots` is not really needed because the command generator could issue a SetSlot command before all of these steps if the user message contains some information already.

Reply:
you don't need to do the change in an action, you can also modify the stack from the flow policy

Reply:
basically, where you are currently updating the slot, you can also just update the stack frame. you don't even need to persist the updated stack as that is done anyways by the policy if there are any changes.

Reply:
Morning @tmbo I'm trying to test the new suggestions after merging the updated `dm2` into this branch. I have the latest `dm2` on `rasa-plus` too, however, when I tried to train the financial demo bot I ran into this error:
```
ModuleNotFoundError: No module named 'rasa.cdu.dialogue_stack'
```
Which branch should I use on rasa-plus side if not dm2, or is this branch being updated soon?